doc/api/os.md

```diff
@@ -49,6 +49,18 @@ Returns an object containing commonly used operating system specific constants
 for error codes, process signals, and so on. The specific constants currently
 defined are described in [OS Constants](#os_os_constants_1).
 
+## os.constrainedmem()
```

Comment:
I think it should be `os.constrainedMemory()`. It's only a few characters longer, but more readable.

This is also more in line with the libuv method which is `uv_get_constrained_memory`, making this a camel-cased version.

I'm aware it's like this for consistency with existing methods, but it's an awful naming convention and I think `os.freemem()` and `os.totalmem()` should be renamed too.

Reply:
While I share the sentiment, that is mostly orthogonal to this PR, I think?

Renaming freemem/totalmem and doc-deprecating the old variants should be proposed as a separate PR, imo, and
this should use the current convention if landed before that PR (and that PR should then also include renaming of this method) and the new convention if landed after that PR.